### ISSUE TITLE
[Bug]: Cannot connect to Web3Auth module, Invalid hooks call

### ISSUE THREAD

**Issue by slotZero01** [blocknative/web3-onboard#1590]
### Current Behavior

There's an issue on connecting with web3auth module, there's an error log of 

```
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

### What package is effected by this issue?

@web3-onboard/web3auth

### Is this a build or a runtime issue?

N/A

### Package Version

2.1.4

### Relevant log output

```shell
Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

____
The above error occurred in the <Modal> component:

    at Modal
```

### Sanity Check

- [X] If this is a build issue, I have included my build config. If this is a runtime issue, I have included reproduction steps and/or a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).

**Comment by Adamj1232** [comment]
@slotZero01 thanks for reporting. 
Can you please provide more info, like what is th version of the @web3-onboard core/react/vue package you are using?
What framework are you using?

**Comment by slotZero01** [comment]
this is all the web3 onboard library that I'm using

```
   "@web3-onboard/coinbase": "^2.2.0",
    "@web3-onboard/core": "^2.15.5",
    "@web3-onboard/injected-wallets": "^2.7.0",
    "@web3-onboard/react": "^2.6.2",
    "@web3-onboard/walletconnect": "^2.3.2",
    "@web3-onboard/web3auth": "^2.1.4",
```

I'm using React V17, the issue only happens when I connect to Web3Auth, the other modules are working properly

**Comment by Adamj1232** [comment]
@slotZero01 that is helpful, thank you! 
FYI You shouldn't need "@web3-onboard/core" if you are using "@web3-onboard/react" as the react package wraps and delivers core packages. May help with package weight and avoiding any future misalignments of core versioning if that were to ever happen. I would recommend removing "@web3-onboard/core" from your package.json 
I will get a web3auth audit into our backlog

**Comment by Adamj1232** [comment]
@slotZero01 please have a look here at the official web3Auth docs - https://web3auth.io/docs/troubleshooting/webpack-issues#react

**Comment by slotZero01** [comment]
No, it's not a webpack issue, that's a compilation issue that I already resolved according to their docs. This is a runtime issue where I cannot connect when clicking to web3Auth button

**Comment by Adamj1232** [comment]
We have updated the web3Auth package to the latest version of the deps and is available at version `2.2.0-alpha.1` Please give this a test.

**Comment by slotZero01** [comment]
I got a different error, `Class extends value undefined is not a constructor or null`

<img width="782" alt="Screenshot 2023-03-27 at 9 48 30 AM" src="https://user-images.githubusercontent.com/121932315/227821656-9d88b358-b631-49a1-958b-8795d5ff94bc.png">

**Comment by Adamj1232** [comment]
@slotZero01 we have this working on a react demo example. Can you please have a look here: https://github.com/blocknative/react-demo/tree/develop and let me know if you find any mis-alignments
